exonum/src/blockchain/config.rs

```diff
@@ -295,6 +296,130 @@ impl ValidateInput for ConsensusConfig {
     }
 }
 
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    ProtobufConvert,
+    BinaryValue,
+    ObjectHash,
+)]
+#[protobuf_convert(source = "runtime::GenesisConfig")]
+pub struct GenesisConfig {
+    /// Blockchain configuration used to create the genesis block.
+    pub consensus_config: ConsensusConfig,
+
+    /// Artifacts specification of builtin services.
+    pub artifacts: Vec<ArtifactSpec>, // TODO: HashSet? Duplications? Perhaps check in builder?
+
+    /// List of the privileged services with the configuration parameters that are created directly
+    /// in the genesis block.
+    pub builtin_instances: Vec<ConfiguredInstanceSpec>,
+}
+
+/// Represents data that is required for initialization of service instance.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    ProtobufConvert,
+    BinaryValue,
+    ObjectHash,
+)]
+#[protobuf_convert(source = "runtime::ConfiguredInstanceSpec")]
+// TODO: Think about better name and/or substitution the `InstanceSpec`
+pub struct ConfiguredInstanceSpec {
+    /// Wrapped `InstanceSpec`.
+    pub instance_spec: InstanceSpec,
+    /// Constructor argument for specific `InstanceSpec`.
+    pub constructor: Vec<u8>,
+}
+
+#[derive(Debug)]
+pub struct GenesisConfigBuilder {
+    /// Consensus config.
+    consensus_config: ConsensusConfig,
+    /// Artifacts specifications for builtin services.
+    artifacts: Vec<ArtifactSpec>,
+    /// Instances of builtin services.
+    builtin_instances: Vec<ConfiguredInstanceSpec>,
+}
+
+#[derive(Debug)]
+pub struct InstanceConfig {
+    pub artifact_spec: ArtifactSpec,
+    pub instances: Vec<ConfiguredInstanceSpec>,
+}
+
+impl InstanceConfig {
+    pub fn new(artifact_id: ArtifactId, deploy_args: Vec<u8>) -> Self {
+        let artifact_spec = ArtifactSpec {
+            artifact: artifact_id,
+            payload: deploy_args,
+        };
+        Self {
+            artifact_spec,
+            instances: vec![],
+        }
+    }
+
+    pub fn with_instance(
+        mut self,
+        instance_spec: InstanceSpec,
+        constructor: impl BinaryValue,
+    ) -> Self {
+        assert_eq!(self.artifact_spec.artifact, instance_spec.artifact);
+        self.instances.push(ConfiguredInstanceSpec {
+            instance_spec,
+            constructor: constructor.into_bytes(),
+        });
+        self
+    }
+}
+
+impl GenesisConfigBuilder {
+    pub fn with_consensus_config(consensus_config: ConsensusConfig) -> Self {
+        Self {
+            consensus_config,
+            artifacts: vec![],
+            builtin_instances: vec![],
+        }
+    }
+
+    pub fn with_builtin_instance(mut self, instance_config: InstanceConfig) -> Self {
```

Comment:
Nit: As far as I understand, right now this method adds *several* instances in general. How often is creating multiple instances of the same service will be performed? (I'd wager, not very.) Thus, it could make sense to make this method add a single service instance, which is presumably returned in a conventional way (e.g., via `Factory::default_instance()` as I've commented above).